integration-tests/bats/stash.bats

```diff
@@ -0,0 +1,295 @@
+
+    dolt sql -q "INSERT INTO test VALUES (3, 'c')"
+    dolt stash
+
+    run dolt stash list
+    [ "$status" -eq 0 ]
+    [ "${#lines[@]}" -eq 3 ]
+
+    run dolt stash clear
+    [ "$status" -eq 0 ]
+
+    run dolt stash list
+    [ "$status" -eq 0 ]
+    [ "${#lines[@]}" -eq 0 ]
+}
+
+@test "stash: stash drop remove an entry at given index in stash list" {
+    skiponoldformat
+    dolt sql -q "INSERT INTO test VALUES (1, 'a')"
+    run dolt stash
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "Created table" ]] || false
+
+    dolt sql -q "INSERT INTO test VALUES (2, 'b')"
+    dolt commit -am "Added row 2 b"
+
+    dolt sql -q "INSERT INTO test VALUES (3, 'c')"
+    run dolt stash
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "Added row 2 b" ]] || false
+
+    run dolt stash list
+    [ "$status" -eq 0 ]
+    [ "${#lines[@]}" -eq 2 ]
+
+    dolt sql -q "INSERT INTO test VALUES (4, 'd')"
+    dolt commit -am "Added row 4 d"
+
+    dolt sql -q "INSERT INTO test VALUES (5, 'e')"
+    run dolt stash
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "Added row 4 d" ]] || false
+
+    run dolt stash list
+    [ "$status" -eq 0 ]
+    [ "${#lines[@]}" -eq 3 ]
+    [[ "${lines[0]}" =~ "Added row 4 d" ]] || false
+    [[ "${lines[1]}" =~ "Added row 2 b" ]] || false
+    [[ "${lines[2]}" =~ "Created table" ]] || false
+
+    run dolt stash drop stash@{1}
+    [ "$status" -eq 0 ]
+
+    run dolt stash list
+    [ "$status" -eq 0 ]
+    [ "${#lines[@]}" -eq 2 ]
+    [[ "${lines[0]}" =~ "Added row 4 d" ]] || false
+    [[ "${lines[1]}" =~ "Created table" ]] || false
+    [[ ! "$output" =~ "Added row 2 b" ]] || false
+}
+
+@test "stash: popping stash on dirty working set with no conflict" {
+    skiponoldformat
+    dolt sql -q "INSERT INTO test VALUES (1, 'a')"
+    run dolt stash
+    [ "$status" -eq 0 ]
+
+    dolt sql -q "INSERT INTO test VALUES (2, 'b')"
+    run dolt sql -q "SELECT * FROM test" -r csv
+    [ "$status" -eq 0 ]
+    [[ ! "$output" = "1,a" ]] || false
+
+    run dolt stash pop 0
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "Dropped refs/stash@{0}" ]] || false
+
+    run dolt sql -q "SELECT * FROM test" -r csv
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "1,a" ]] || false
+    [[ "$output" =~ "2,b" ]] || false
+}
+
+@test "stash: popping stash on dirty working set with conflict" {
+    skiponoldformat
+    dolt sql -q "INSERT INTO test VALUES (1, 'a')"
+    run dolt stash
+    [ "$status" -eq 0 ]
+
+    dolt sql -q "INSERT INTO test VALUES (1, 'b')"
+    run dolt stash pop
+    [ "$status" -eq 1 ]
+    [[ "$output" =~ "error: Your local changes to the following tables would be overwritten by merge" ]] || false
+    [[ "$output" =~ "The stash entry is kept in case you need it again." ]] || false
+
+    run dolt sql -q "SELECT * FROM test" -r csv
+    [[ "$output" =~ "1,b" ]] || false
+    [[ ! "$output" =~ "1,a" ]] || false
+
+    run dolt stash list
+    [ "$status" -eq 0 ]
+    [ "${#lines[@]}" -eq 1 ]
+}
+
+skiponoldformat() {
```

Comment:
You could reuse the `skip_nbf_not_dolt()` function from `common.bash` and avoid duplicating this.

Reply:
I did not use it because it also skips `__DOLT_DEV__`, which we can still test since it's I think using new format. I moved this function into `common.bash` so it's usable in other bats tests as well. 🙂